go/gossip/handler.go

```diff
@@ -66,13 +85,29 @@ func (h *Handler) HandleSTHPollination(rw http.ResponseWriter, req *http.Request
 		return
 	}
 
+	sthToKeep := make([]ct.SignedTreeHead, 0, len(p.STHs))
+	for _, sth := range p.STHs {
+		v, found := h.verifiers[sth.LogID]
+		if !found {
+			log.Printf("Pollination entry for unknown logID: %s", sth.LogID.Base64String())
+			continue
+		}
+		if err := v.VerifySTHSignature(sth); err != nil {
+			log.Printf("Failed to verify STH, dropping: %v", err)
+			continue
+		}
+		sthToKeep = append(sthToKeep, sth)
+	}
+	p.STHs = sthToKeep
```

Comment:
'cos I'm actually, in effect, *removing* things from` p.STHs` here, and using the temporary this way so it's `O(n)` and not `(n*n)` in element copies. Could do that in-place by swapping elements and truncating of course, but that seems to be extra complexity for no real gain.